servicex/configuration.py

```diff
@@ -46,6 +47,35 @@ class Configuration(BaseModel):
     cache_path: Optional[str] = Field(alias="cache-path", default=None)
     shortened_downloaded_filename: Optional[bool] = False
 
+    @validator("cache_path", always=True)
+    def expand_cache_path(cls, v):
+        """
+        Expand the cache path to a full path, and create it if it doesn't exist.
+        Expand ${USER} to be the user name on the system. Works for windows, too.
+        :param v:
+        :return:
+        """
+        if not v:
+            v = "/tmp"
+
+        s_path = os.path.expanduser(v)
+
+        # If they have tried to use the USER or UserName as an expansion, and it has failed, then
+        # translate it to maintain harmony across platforms.
+        if "${USER}" in s_path and "UserName" in os.environ:
+            s_path = s_path.replace("${USER}", os.environ["UserName"])
+        if "${USER}" in s_path and "USER" in os.environ:
+            s_path = s_path.replace("${USER}", os.environ["USER"])
+
+        p_p = PurePath(s_path)
+        if len(p_p.parts) > 1 and p_p.parts[1] == "tmp":
```

Comment:
@gordonwatts this line replaces `/tmp` with the correct temporary directory for the platform.